curp/src/server/raw_curp/mod.rs

```diff
@@ -541,6 +550,22 @@ impl<C: Command, RC: RoleChange> RawCurp<C, RC> {
         Ok(())
     }
 
+    /// Handle `lease_keep_alive` message
+    pub(super) fn handle_lease_keep_alive(&self, client_id: u64) -> Option<u64> {
+        if client_id == 0 {
+            return Some(self.ctx.lm.write().grant());
+        }
+        self.ctx.lm.map_write(|mut lm_w| {
+            if lm_w.check_alive(client_id) {
+                lm_w.renew(client_id);
+                None
+            } else {
+                lm_w.revoke(client_id);
+                Some(lm_w.grant())
+            }
+        })
```

Comment:
All operations in this method require a write lock of lease_manager, and I recommend getting a write lock directly at the beginning of the method to reduce nesting

Reply:
done